op-challenger/game/fault/trace/asterisc/provider.go

```diff
@@ -180,21 +172,21 @@ type AsteriscTraceProviderForTest struct {
 	*AsteriscTraceProvider
 }
 
-func NewTraceProviderForTest(logger log.Logger, m AsteriscMetricer, cfg *config.Config, localInputs utils.LocalGameInputs, dir string, gameDepth types.Depth) *AsteriscTraceProviderForTest {
+func NewTraceProviderForTest(logger log.Logger, m vm.Metricer, cfg *config.Config, localInputs utils.LocalGameInputs, dir string, gameDepth types.Depth) *AsteriscTraceProviderForTest {
 	p := &AsteriscTraceProvider{
 		logger:         logger,
 		dir:            dir,
 		prestate:       cfg.AsteriscAbsolutePreState,
-		generator:      NewExecutor(logger, m, cfg, cfg.AsteriscNetwork, localInputs),
+		generator:      vm.NewExecutor(logger, m, cfg.Asterisc, cfg.Asterisc.Network, localInputs),
```

Comment:
```suggestion
		generator:      vm.NewExecutor(logger, m, cfg.Asterisc, cfg.AsteriscAbsolutePreState, localInputs),
```
Based on https://github.com/ethereum-optimism/optimism/pull/10774. Maybe we need a rebase.